pallets/parachain-staking/src/lib.rs

```diff
@@ -2859,13 +2900,22 @@ pub mod pallet {
 				DispatchClass::Mandatory,
 			);
 
-			let collators = Pallet::<T>::selected_candidates().to_vec();
-			if collators.is_empty() {
+			let selected_candidates = Pallet::<T>::selected_candidates().to_vec();
+			Self::prepare_delayed_rewards(&selected_candidates, new_index);
+
+			if selected_candidates.is_empty() {
 				// we never want to pass an empty set of collators. This would brick the chain.
 				log::error!("💥 keeping old session because of empty collator set!");
+
+				// get collators of previous session for snapshot
+				let old_collators = AtStake::<T>::iter_prefix(new_index - 1)
```

Comment:
I remember that last time you mentioned that in some rare situations (during the runtime upgrade and encounter, the AtStake is zero), do you think we will handle this case?

Reply:
So this case can only be problematic in runtime upgrade, we fixed that by doing a snapshot of active collators at runtime upgrade with a storage migration so they'll get delayed rewards in new session that we force start. This way, unless somehow, there were no candidates before runtime upgrade, we cannot run into this issue.